ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -482,6 +513,19 @@ chainSyncClient mkPipelineDecision0 tracer cfg
                 (Their theirTip')
         }
 
+    -- | Update the candidate var or in case the server lies about their tip,
+    -- disconnect with a 'InvalidTip' exception.
+    updateCandidateVar :: Their (Tip blk)
+                       -> AnchoredFragment (Header blk)
+                       -> m ()
+    updateCandidateVar theirTip theirFrag =
+      case AF.isFragmentComplete (castTip $ unTheir theirTip) theirFrag of
+         Nothing -> disconnect $
```

Comment:
It is not clear to my why seeing an incomplete fragment in this context (which I assume that's what this `Nothing` represents) means that the server lies about its tip.